ts/src/xt.ts

```diff
@@ -0,0 +1,4066 @@
+        //         "price": "27368",
+        //         "fee": "0.02134704",
+        //         "feeCoin": "usdt",
+        //         "timestamp": 1679116769838,
+        //         "takerMaker": "TAKER"
+        //     }
+        //
+        const marketId = this.safeString2 (trade, 's', 'symbol');
+        let marketType = (market !== undefined) ? market['type'] : undefined;
+        if (marketType === undefined) {
+            marketType = ('b' in trade) || ('bizType' in trade) ? 'spot' : 'contract';
+        }
+        market = this.safeMarket (marketId, market, '_', marketType);
+        const bidOrAsk = this.safeString (trade, 'm');
+        let side = this.safeStringLower (trade, 'orderSide');
+        if (bidOrAsk !== undefined) {
+            side = (bidOrAsk === 'BID') ? 'buy' : 'sell';
+        }
+        const buyerMaker = this.safeValue (trade, 'b');
+        let takerOrMaker = this.safeStringLower (trade, 'takerMaker');
+        if (buyerMaker !== undefined) {
+            takerOrMaker = buyerMaker ? 'maker' : 'taker';
+        }
+        const timestamp = this.safeIntegerN (trade, [ 't', 'time', 'timestamp' ]);
+        const quantity = this.safeString2 (trade, 'q', 'quantity');
+        const amount = (marketType === 'spot') ? quantity : Precise.stringMul (quantity, this.numberToString (market['contractSize']));
+        return {
+            'info': trade,
+            'id': this.safeStringN (trade, [ 'i', 'tradeId', 'execId' ]),
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'symbol': market['symbol'],
+            'order': this.safeString (trade, 'orderId'),
+            'type': this.safeStringLower (trade, 'orderType'),
+            'side': side,
+            'takerOrMaker': takerOrMaker,
+            'price': this.safeNumber2 (trade, 'p', 'price'),
+            'amount': this.parseNumber (amount),
+            'cost': undefined,
+            'fee': {
+                'currency': this.safeCurrencyCode (this.safeString2 (trade, 'feeCurrency', 'feeCoin')),
+                'cost': this.safeNumber (trade, 'fee'),
+                'rate': undefined,
+            },
+        };
+    }
+
+    async fetchBalance (params = {}) {
+        /**
+         * @method
+         * @name xt#fetchBalance
+         * @description query for balance and get the amount of funds available for trading or funds locked in orders
+         * @see https://doc.xt.com/#balancebalancesGet
+         * @see https://doc.xt.com/#futures_usergetBalances
+         * @param {object} params extra parameters specific to the xt api endpoint
+         * @returns {object} a [balance structure]{@link https://docs.ccxt.com/en/latest/manual.html?#balance-structure}
+         */
+        await this.loadMarkets ();
+        let subType = undefined;
+        [ subType, params ] = this.handleSubTypeAndParams ('fetchBalance', undefined, params);
+        let response = undefined;
+        if (subType === 'linear') {
```

Comment:
can we also support `type=swap` here?

Reply:
These endpoints return future and swap, should I support ` if (type = 'swap' or type = 'future')` ?

Reply:
yes, in the case of swap we should default to `linear` if `subtType` is not available